rmw_fastrtps_cpp/src/TypeSupport.cpp

```diff
@@ -202,8 +202,10 @@ bool TypeSupport::serializeROSmessage(eprosima::fastcdr::Cdr &ser, const rosidl_
             {
                 case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_BOOL:
                     {
-                        bool sb = false, &b = *(bool*)field;
-                        if(b) sb = true; 
+                        bool sb = false;
+                        // don't cast to bool here because if the bool is
+                        // uninitialized the random value can't be deserialized
+                        if(*(uint8_t*)field) sb = true;
```

Comment:
What the compiler does is basically optimize this line to look like this:

```
// if(b) sb = true;
// since the compiler "knows" that b is true
// it generates assembler code reusing the already loaded value
if(b) sb = b;
```